src/firefly/js/visualize/ui/DrawLayerUIComponents.jsx

```diff
@@ -50,3 +56,55 @@ export function getMinMaxWidth(maxTitleChars) {
     const maxW = maxTitleChars*.7 > 10 ? Math.min(maxTitleChars*.7, 30) : 10;
     return {minW,maxW};
 }
+
+export function TableColorTitle({color,drawLayerId, plotId, tbl_id}) {
+
+    const table= getTblById(tbl_id);
+    const dl= getDrawLayerById(getDlAry(), drawLayerId);
+    if (!table || !dl) return;
+
+    return (
+        <Stack direction='row' alignItems='center' spacing={1/4} overflow='hidden'>
+            <Button sx={{p:'3px', minHeight:0, zIndex:2}}
+                onClick={() => modifyDrawColor(dl,plotId,tbl_id)}>
+                <Box {...{
+                     width:symbolSize, height:symbolSize, backgroundColor: color,
+                    border:'1px solid transparent', borderRadius:3
+                }} />
+            </Button>
+            <Typography level='body-sm' noWrap={true} title={table.title}>{truncate(table.title)}</Typography>
+        </Stack>
+    );
+}
+
+export function makeTableColorTitle(color, drawLayerId, plotId, tbl_id) {
+    return <TableColorTitle {...{color, drawLayerId, plotId, tbl_id}} />;
+}
+
+export function modifyDrawColor(inDl, plotId, tbl_id) {
+    hideColorPickerDialog();
+    showColorPickerDialog(inDl.drawingDef.color, inDl.canUserChangeColor === ColorChangeType.STATIC, false,
+        (ev) => {
+            const {r, g, b, a} = ev.rgb;
+            const rgbStr = `rgba(${r},${g},${b},${a})`;
+
+            if (tbl_id && inDl.tableCanControlColor) {
+                const dlAryForTable= getDlAry().filter( (dl) => tbl_id===dl.tbl_id && dl.tableCanControlColor);
+                dlAryForTable.forEach( (dl) => {
+                    dispatchChangeDrawingDef(dl.displayGroupId, Object.assign({}, dl.drawingDef, {color: rgbStr}), plotId, dl.titleMatching);
+                    plotId= dl.plotIdAry?.[0];
+                    const {tbl_ui_id} = getTableUiByTblId(tbl_id) ?? {};
+                    if (!tbl_ui_id && !plotId) return;
+                    dispatchTableUiUpdate({tbl_ui_id,
+                        title:makeTableColorTitle(rgbStr,dl.drawLayerId,plotId,tbl_id),
+                        color: rgbStr
+                    });
+                });
+            }
+            else {
+                const dl = getDrawLayersByDisplayGroup(getDlAry(), inDl.displayGroupId);
+                dispatchChangeDrawingDef(dl.displayGroupId, Object.assign({}, dl.drawingDef, {color: rgbStr}), plotId, dl.titleMatching);
+            }
+            ev.stopPropagation?.();
```

Comment:
Any reason for stopping propagation? I thought it would have stopped changing tabs if I clicked on color swatch of a tab that's not active but that doesn't happen in the UI, so just curious.

Reply:
Probably not. I think it has been this week with this component for years. I refactored the call `showColorPickerDialog` from another place. I will experiment with removing it.